lib/AST/FeatureSet.cpp

```diff
@@ -506,6 +506,29 @@ UNINTERESTING_FEATURE(Embedded)
 UNINTERESTING_FEATURE(Volatile)
 UNINTERESTING_FEATURE(SuppressedAssociatedTypes)
 
+static bool disallowFeatureSuppression(StringRef featureName, Decl *decl);
+
+static bool allBoundTypesAreCopyable(Type type, DeclContext *context) {
+  assert(type->getAnyNominal());
+  auto bgt = type->getAs<BoundGenericType>();
+  if (!bgt)
+    return false;  // nothing is bound.
+
+  for (auto argInterfaceTy : bgt->getGenericArgs()) {
+    auto argTy = context->mapTypeIntoContext(argInterfaceTy);
+    if (!argTy)
+      return false; // assume the worst: it's noncopyable
+
+    if (argTy->hasTypeParameter() || argTy->hasUnboundGenericType())
```

Comment:
The first condition can’t happen. The second shouldn’t happen because there is insufficient information, so you should assert against it.